src/hidpp20.h

```diff
@@ -317,6 +324,46 @@ union hidpp20_button_binding {
 } __attribute__((packed));
 _Static_assert(sizeof(union hidpp20_button_binding) == 4, "Invalid size");
 
+struct hidpp20_color {
+	uint8_t red;
+	uint8_t green;
+	uint8_t blue;
+};
+
+// TODO: put the leds into array and access them through this enum
+enum hidpp20_led_type {
+	HIDPP20_LED_UNKNOWN = -1,
+	HIDPP20_LED_LOGO = 0,
+	HIDPP20_LED_SIDE
+};
+
+struct hidpp20_internal_led {
+	uint8_t mode;                  /* 00 - off; 01 - on; 03 - cycle; 0a - breath */
+	struct hidpp20_color color;
+	union {
+		struct {
+			uint8_t free[2];
+			uint16_t rate;          /* 100 - 20000 */
+			uint8_t bright;         /* 1 - 100 */
+			uint8_t free2[2];
+		} __attribute__((packed)) cycle;
+		struct hidpp20_led_breath {
+			uint16_t rate;
+			uint8_t free;
+			uint8_t bright;
+			uint8_t free2[3];
+		} __attribute__((packed)) breath;
+	} __attribute__((packed)) effect;
+};
+_Static_assert(sizeof(struct hidpp20_internal_led) == 11, "Invalid size");
+
+struct hidpp20_led {
+	enum hidpp20_led_mode mode;
+	struct hidpp20_color color;
+	uint16_t rate;
+	uint8_t bright;
+} __attribute__((packed));
+
```

Comment:
generally, wherever you have a packed struct you should add a static assert to make sure it works as expected. the only reason to pack a struct is for it to go on the wire, so better to be doubly safe here.

I"m wondering why this one is packed though, it seems like a normal struct?